pulsar-websocket/src/main/java/org/apache/pulsar/websocket/ProducerHandler.java

```diff
@@ -315,30 +351,117 @@ protected ProducerBuilder<byte[]> getProducerBuilder(PulsarClient client) {
+        try {
+            param = Base64.getDecoder().decode(StringUtils.trim(queryParams.get("encryptionKeys")));
+        } catch (Exception base64DecodeEx) {
+            log.error("Could not base64 decode the param encryptionKeys", base64DecodeEx);
+            throw new IllegalArgumentException("Could not base64 decode the param encryptionKeys");
+        }
+        try {
+            Map<String, EncryptionKey> keys = ObjectMapperFactory.getMapper().getObjectMapper()
+                    .readValue(param, new TypeReference<Map<String, EncryptionKey>>() {
+                    });
+            if (keys.isEmpty()) {
+                return null;
+            }
+            if (keys.values().iterator().next().getKeyValue() == null) {
+                return null;
+            }
+            return keys;
+        } catch (IOException ex) {
+            return null;
         }
+    }
 
-        if (queryParams.containsKey("batchingMaxPublishDelay")) {
-            builder.batchingMaxPublishDelay(Integer.parseInt(queryParams.get("batchingMaxPublishDelay")),
-                    TimeUnit.MILLISECONDS);
+    private void popularProducerBuilderForClientSideEncrypt(ProducerBuilder<byte[]> builder,
+                                                            Map<String, EncryptionKey> encryptionKeyMap) {
+        this.clientSideEncrypt = true;
+        int keysLen = encryptionKeyMap.size();
+        final String[] keyNameArray = new String[keysLen];
+        final byte[][] keyValueArray = new byte[keysLen][];
+        final List<KeyValue>[] keyMetadataArray = new List[keysLen];
+        // Format keys.
+        int index = 0;
+        for (Map.Entry<String, EncryptionKey> entry : encryptionKeyMap.entrySet()) {
+            checkArgument(StringUtils.isNotBlank(entry.getKey()), "Empty param encryptionKeys.key");
+            checkArgument(entry.getValue() != null, "Empty param encryptionKeys.value");
+            checkArgument(entry.getValue().getKeyValue() != null, "Empty param encryptionKeys.value.keyValue");
+            keyNameArray[index] = StringUtils.trim(entry.getKey());
+            keyValueArray[index] = entry.getValue().getKeyValue();
+            if (entry.getValue().getMetadata() == null) {
+                keyMetadataArray[index] = Collections.emptyList();
+            } else {
+                keyMetadataArray[index] = entry.getValue().getMetadata().entrySet().stream()
+                        .map(e -> new KeyValue().setKey(e.getKey()).setValue(e.getValue()))
+                        .collect(Collectors.toList());
+            }
+            builder.addEncryptionKey(keyNameArray[index]);
         }
+        // Disable server-side batch process, compression, and encryption, and only set the message metadata that
+        // which client-side provided.
+        builder.enableBatching(false);
+        builder.compressionType(CompressionType.NONE);
+        builder.cryptoKeyReader(DummyCryptoKeyReaderImpl.INSTANCE);
```

Comment:
Please add a comment for these two lines to explain why we are using NONE compressor and DummyCryptoKeyReader.

Reply:
Done